testing/python_lib/test_failscale.py

```diff
@@ -35,23 +34,55 @@ def __init__(self, *args, **kwargs):
 
     def test_stack_connectivity(self):
         """Test to build stack and check for connectivity"""
+
+        faux_args = []
+        for dev_index in range(self.devices):
+            for sw_index in range(1, self.switches + 1):
+                device_num = dev_index * self.switches + sw_index
+                switch = 't2sw%s' % sw_index
+                device_port = 100 + dev_index + 1
+                faux_args.append((switch, device_port, device_num))
+
+        self.parallelize(len(faux_args), self.add_faux, faux_args)
+
+        time.sleep(10)
+
         device_list = ['forch-faux-'+str(num) for num in range(1, self.devices * self.switches + 1)]
+
+        def ping_device(device, ping_count):
+            ping_count[0] = self._ping_host(device, '192.168.1.0', count=10, output=True)
+            ping_count[1] = self._ping_host(device, '192.168.1.1', count=10, output=True)
+
+        ping_counts = {}
+        target_args = []
         for device in device_list:
-            self.assertLessEqual(1, self._ping_host(device, '192.168.1.0', count=3, output=True),
-                                 'Couldn\'t reach 192.168.1.0 from %s' % device)
-            self.assertLessEqual(1, self._ping_host(device, '192.168.1.1', count=3, output=True),
-                                 'Couldn\'t reach 192.168.1.1 from %s' % device)
+            # Get shared memory int array which ensures same copy of the array across
+            # multiple processes
```

Comment:
if it's shared across multiple processes why is it allocated once for each one?  Something about this doesn't smell right, so either it's confused, or I'm confused. If we can't figure it out in the comments then maybe we'll need a quick GVC to clarify...